mojo/core/channel_posix.cc

```diff
@@ -116,6 +116,21 @@ class ChannelPosix : public Channel,
     CHECK(server_.is_valid() || socket_.is_valid());
   }
 
+  void SetSocket(ConnectionParams connection_params) override {
+    read_watcher_.reset();
+    write_watcher_.reset();
+
+    ignore_result(socket_.release());
+    server_.TakePlatformHandle().release();
+
+    socket_ = connection_params.TakeEndpoint().TakePlatformHandle().TakeFD();
+  }
+
+  void ClearOutgoingMessages() override {
+    base::circular_deque<MessageView> messages;
+    std::swap(outgoing_messages_, messages);
```

Comment:
Isn't it just okay to call outgoing_messages_.clear()?

Reply:
Ok, I will change this as you pointed out.